protocol-units/bridge/chains/movement/src/lib.rs

```diff
@@ -520,33 +530,93 @@ impl BridgeContractCounterparty for MovementClient {
 		bridge_transfer_id: BridgeTransferId<Self::Hash>,
 	) -> BridgeContractCounterpartyResult<()> {
 		let args = vec![
-			utils::serialize_vec(&self.signer.address().to_vec())?,
 			utils::serialize_vec(&bridge_transfer_id.0)?,
 		];
 		let payload = utils::make_aptos_payload(
 			self.counterparty_address,
 			COUNTERPARTY_MODULE_NAME,
 			"abort_bridge_transfer",
-			self.counterparty_type_args(Call::Abort),
+			Vec::new(),
 			args,
 		);
-		let _ = utils::send_and_confirm_aptos_transaction(
+		let result = utils::send_and_confirm_aptos_transaction(
 			&self.rest_client,
 			self.signer.as_ref(),
 			payload,
 		)
 		.await
 		.map_err(|_| BridgeContractCounterpartyError::AbortTransferError);
+
+		println!("Abort bridge transfer result: {:?}", &result);
 		Ok(())
 	}
 
 	async fn get_bridge_transfer_details(
 		&mut self,
-		_bridge_transfer_id: BridgeTransferId<Self::Hash>,
-	) -> BridgeContractCounterpartyResult<Option<BridgeTransferDetails<Self::Address, Self::Hash>>>
-	{
-		todo!();
-	}
+		bridge_transfer_id: BridgeTransferId<[u8; 32]>,
+	    ) -> Result<Option<BridgeTransferDetails<MovementAddress, [u8; 32]>>, BridgeContractCounterpartyError> {
+		// Convert the bridge_transfer_id to a hex string
+		let bridge_transfer_id_hex = format!("0x{}", hex::encode(bridge_transfer_id.0));
+	    
+		// Construct the ViewRequest
+		let view_request = ViewRequest {
+		    function: EntryFunctionId {
+			module: MoveModuleId {
+			    address: self.counterparty_address.clone().into(),
+			    name: aptos_api_types::IdentifierWrapper(Identifier::new("atomic_bridge_counterparty")
+				.map_err(|_| BridgeContractCounterpartyError::FunctionViewError)?),
+			},
+			name: aptos_api_types::IdentifierWrapper(Identifier::new("bridge_transfers")
+			    .map_err(|_| BridgeContractCounterpartyError::FunctionViewError)?),
+		    },
+		    type_arguments: vec![],
+		    arguments: vec![serde_json::json!(bridge_transfer_id_hex)],
+		};
+	    
+		// Send the request to the "/view" endpoint using JSON
+		let response: Response<Vec<serde_json::Value>> = self.rest_client
+		    .view(&view_request, None)
+		    .await
+		    .map_err(|_| BridgeContractCounterpartyError::CallError)?;
+	    
+		// Extract and parse the response
+		let values = response.inner();
+		if values.len() == 6 {
+		    let originator = values[0].as_str().ok_or(BridgeContractCounterpartyError::SerializationError)?;
+		    let recipient = values[1].as_str().ok_or(BridgeContractCounterpartyError::SerializationError)?;
+		    let amount = values[2].as_str().ok_or(BridgeContractCounterpartyError::SerializationError)?.parse::<u64>().map_err(|_| BridgeContractCounterpartyError::SerializationError)?;
```

Comment:
Getting the value by index [0] can panic, I know you are handling the error and sure that these vals exist but its usually better to use `values.get(1)`.